### PR TITLE
Add `grepText` to the Prelude

### PR DESCRIPTION
**Pull request by markus1189**
During my scripting, I missed a function that could `grep` a `Shell Text` and so I wrote one ;)

I added an internal helper `grepWith` that is parameterized over an additional function wich converts to `Text` and used that one from both `grep` and `grepText`.

### CODE REVIEW

File: src/Turtle/Prelude.hs [hunk 1]
```diff
@@ -1513,12 +1514,16 @@ appendonly file = using (managed (Filesystem.withTextFile file IO.AppendMode))
 cat :: [Shell a] -> Shell a
 cat = msum
 
+grepWith :: (b -> Text) -> Pattern a -> Shell b -> Shell b
+grepWith f pattern s = mfilter (not . null . match pattern . f) s
```
Comment (markus1189) [inline src/Turtle/Prelude.hs]: Was there a reason to *not* use `mfilter` before this?

Comment (Gabriel439) [reply src/Turtle/Prelude.hs]: No reason.  `mfilter` is just fine :)

**Comment by Gabriel439** [comment]
I'll merge this as soon as CI passes

**Comment by markus1189** [comment]
We could also add a `grepFilePath :: Pattern a -> Shell FilePath -> Shell FilePath` which would be just `onFiles . grepText` but I don't know whether it is worth to make that more visible or if users should just write the version with `onFiles`

**Comment by Gabriel439** [comment]
Yeah, I would recommend using `onFiles` for the `FilePath` case

**Comment by Gabriel439** [comment]
I can push a new release to Hackage if you have no other pending changes

**Comment by markus1189** [comment]
You are really super fast, I appreciate that.  Let's wait one or two days, I might find more stuff to add and I don't mind having it defined inside my script for now ;)

**Comment by markus1189** [comment]
You are really super fast, I appreciate that.  Let's wait one or two days, I might find more stuff to add and I don't mind having it defined inside my script for now ;)

**Comment by markus1189** [comment]
You are really super fast, I appreciate that.  Let's wait one or two days, I might find more stuff to add and I don't mind having it defined inside my script for now ;)

**Comment by markus1189** [comment]
You are really super fast, I appreciate that.  Let's wait one or two days, I might find more stuff to add and I don't mind having it defined inside my script for now ;)

**Pull request merged**